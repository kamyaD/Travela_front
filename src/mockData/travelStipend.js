const mockData = {
  stipends: [
    {
      id:1,
      center: {location: 'Nairobi, Kenya'},
      amount: 345
    },
    {
      id:2,
      center: {location: 'Kampala, Uganda'},
      amount: 345
    }
  ],

  selectedStipend: {
    id: 29,
    amount: 500,
    creator: {fullName:'Mirriam Maina',id:1},
    center: {location:'Kampala, Uganda'}
  }
};

export default mockData;
