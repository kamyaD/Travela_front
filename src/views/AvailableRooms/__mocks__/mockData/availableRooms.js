const beds = [
  {
    id: 6000,
    bedName: 'bed 1',
    booked: false,
    createdAt: '2018-08-16T11:11:52.181Z',
    updatedAt: '2018-08-16T11:11:52.181Z',
    roomId: 'room-id-1',
    rooms: {
      id: 'room-id-1',
      roomName: 'Kwetu',
      roomType: 'ensuited',
      bedCount: 2,
      faulty: false,
      createdAt: '2018-08-16T11:11:52.181Z',
      updatedAt: '2018-08-16T11:11:52.181Z',
      guestHouseId: 'guest-house-1',
      guestHouses: {
        id: 'guest-house-1',
        houseName: 'Mini flat-B',
        location: 'Lagos,Nigeria',
        bathRooms: 1,
        imageUrl:
          'https://www.dropbox.com/s/c2n63x2jamh3ndv/guesthouse2.jpg?raw=1',
        createdAt: '2018-08-16T11:11:52.181Z',
        updatedAt: '2018-08-16T11:11:52.181Z',
        userId: '-LJV4b1QTCYewOtk5F63'
      }
    }
  },
  {
    id: 6002,
    bedName: 'bed 1',
    booked: false,
    createdAt: '2018-08-16T11:11:52.181Z',
    updatedAt: '2018-08-16T11:11:52.181Z',
    roomId: 'room-id-2',
    rooms: {
      id: 'room-id-2',
      roomName: 'Menengai',
      roomType: 'ensuited',
      bedCount: 3,
      faulty: false,
      createdAt: '2018-08-16T11:11:52.181Z',
      updatedAt: '2018-08-16T11:11:52.181Z',
      guestHouseId: 'guest-house-1',
      guestHouses: {
        id: 'guest-house-1',
        houseName: 'Mini flat-B',
        location: 'Lagos,Nigeria',
        bathRooms: 1,
        imageUrl:
          'https://www.dropbox.com/s/c2n63x2jamh3ndv/guesthouse2.jpg?raw=1',
        createdAt: '2018-08-16T11:11:52.181Z',
        updatedAt: '2018-08-16T11:11:52.181Z',
        userId: '-LJV4b1QTCYewOtk5F63'
      }
    }
  },
  {
    id: 6003,
    bedName: 'bed 2',
    booked: false,
    createdAt: '2018-08-16T11:11:52.181Z',
    updatedAt: '2018-08-16T11:11:52.181Z',
    roomId: 'room-id-2',
    rooms: {
      id: 'room-id-2',
      roomName: 'Menengai',
      roomType: 'ensuited',
      bedCount: 3,
      faulty: false,
      createdAt: '2018-08-16T11:11:52.181Z',
      updatedAt: '2018-08-16T11:11:52.181Z',
      guestHouseId: 'guest-house-1',
      guestHouses: {
        id: 'guest-house-1',
        houseName: 'Mini flat-B',
        location: 'Lagos,Nigeria',
        bathRooms: 1,
        imageUrl:
          'https://www.dropbox.com/s/c2n63x2jamh3ndv/guesthouse2.jpg?raw=1',
        createdAt: '2018-08-16T11:11:52.181Z',
        updatedAt: '2018-08-16T11:11:52.181Z',
        userId: '-LJV4b1QTCYewOtk5F63'
      }
    }
  },
  {
    id: 6004,
    bedName: 'bed 3',
    booked: false,
    createdAt: '2018-08-16T11:11:52.181Z',
    updatedAt: '2018-08-16T11:11:52.181Z',
    roomId: 'room-id-2',
    rooms: {
      id: 'room-id-2',
      roomName: 'Menengai',
      roomType: 'ensuited',
      bedCount: 3,
      faulty: false,
      createdAt: '2018-08-16T11:11:52.181Z',
      updatedAt: '2018-08-16T11:11:52.181Z',
      guestHouseId: 'guest-house-1',
      guestHouses: {
        id: 'guest-house-1',
        houseName: 'Mini flat-B',
        location: 'Lagos,Nigeria',
        bathRooms: 1,
        imageUrl:
          'https://www.dropbox.com/s/c2n63x2jamh3ndv/guesthouse2.jpg?raw=1',
        createdAt: '2018-08-16T11:11:52.181Z',
        updatedAt: '2018-08-16T11:11:52.181Z',
        userId: '-LJV4b1QTCYewOtk5F63'
      }
    }
  },
  {
    id: 6001,
    bedName: 'bed 2',
    booked: true,
    createdAt: '2018-08-16T11:11:52.181Z',
    updatedAt: '2018-10-18T17:27:08.163Z',
    roomId: 'room-id-1',
    rooms: {
      id: 'room-id-1',
      roomName: 'Kwetu',
      roomType: 'ensuited',
      bedCount: 2,
      faulty: false,
      createdAt: '2018-08-16T11:11:52.181Z',
      updatedAt: '2018-08-16T11:11:52.181Z',
      guestHouseId: 'guest-house-1',
      guestHouses: {
        id: 'guest-house-1',
        houseName: 'Mini flat-B',
        location: 'Lagos,Nigeria',
        bathRooms: 1,
        imageUrl:
          'https://www.dropbox.com/s/c2n63x2jamh3ndv/guesthouse2.jpg?raw=1',
        createdAt: '2018-08-16T11:11:52.181Z',
        updatedAt: '2018-08-16T11:11:52.181Z',
        userId: '-LJV4b1QTCYewOtk5F63'
      }
    }
  }
];

export default beds;
