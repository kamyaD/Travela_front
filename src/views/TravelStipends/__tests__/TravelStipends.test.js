import React from 'react';
import { mount } from 'enzyme';
import mockData from '../../../mockData/travelStipend';
import sinon from 'sinon';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { TravelStipends, mapStateToProps } from '..';

const { stipends } = mockData;
let props = {
  travelStipends : {
    travelStipends: {},
    errors: [],
    isLoading: false,
    stipends
  },
  centers: {
    center: [
      {
        location: 'kigali'
      },
    ]
  },
  handleCreateTravelStipend: jest.fn(() => {}),
  CreateTravelStipend: jest.fn(() => {}),
  isLoading: false,
  createNewRequest: jest.fn(),
  loading: false,
  errors: [],
  shouldOpen: false,
  fetchCenters: jest.fn(),
  modalType: 'create travel stipend',
  openModal: sinon.spy(() => Promise.resolve()),
  closeModal: sinon.spy(() => Promise.resolve()),
  fetchAllTravelStipends: jest.fn(),
};

const initialState = {
  auth: {
    isAuthenticated: true,
    user: {
      UserInfo: {
        name: 'Tomato Jos',
        picture: 'http://picture.com/gif'
      }
    }
  },
  travelStipends: {
    errors: {},
    stipends,
    isLoading: false
  },
  modalReducer: {
    shouldOpen: false,
    modalType: null
  },
};
const mockStore = configureStore();
const store = mockStore(initialState);

describe('<TravelStipends>', () => {
  it('should render the TravelStipends page without crashing', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <TravelStipends {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.length).toBe(1);
    wrapper.unmount();
  });

  it('renders as expected', () => {
    const wrapper = mount(<TravelStipends {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  
  it('should set `shouldOpen` prop to `true` when add stipend button is clicked', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <TravelStipends {...{...props, shouldOpen: true, modalType: 'create travel stipend'}} />
        </MemoryRouter>
      </Provider>
    );
    wrapper.find('button.action-btn').simulate('click');
    expect(wrapper.find('TravelStipends').props().shouldOpen).toBe(true);
    expect(props.openModal.called).toBe(true);
    expect(
      wrapper
        .find('Modal')
        .at(0)
        .props().visibility
    ).toEqual('visible');
  });

  it('should set `visibility` prop to `visible` when add stipend button is clicked', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <TravelStipends
            {...{ ...props, shouldOpen: true, modalType: 'create travel stipend' }}
          />
        </MemoryRouter>
      </Provider>
    );
    wrapper.find('button.action-btn').simulate('click');
    expect(
      wrapper
        .find('Modal')
        .at(0)
        .props().visibility
    ).toEqual('visible');
  });

  it('maps state to props and return the expected object', () => {
    const travelStipends = {
      travelStipends: {},
    };

    const modal = {
      modal: {
        shouldOpen: false,
        modalType: null
      }
    };

    const centers = {
      center: [
        {
          location: 'kigali'
        }
      ]
    };

    const props = mapStateToProps({travelStipends, modal, centers});
    expect(props).toEqual({
      ...travelStipends, ...modal.modal,
      centers: centers,
      travelStipends: travelStipends,
    });
  });
});
