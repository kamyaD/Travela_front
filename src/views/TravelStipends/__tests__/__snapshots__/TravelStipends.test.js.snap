// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TravelStipends> renders as expected 1`] = `
Array [
  <div
    className="travelStipends--header"
>
    <div
        className="PageHeader"
    >
        <div>
            <span
                className="title"
            >
                TRAVEL STIPEND
            </span>
        </div>
        <div>
            <button
                className="action-btn btn-new-request"
                onClick={[Function]}
                type="button"
            >
                Add Stipend
            </button>
        </div>
    </div>
</div>,
  "",
  <div>
    <div>
        <div>
            <div
                className="card"
            >
                <i
                    className="fa fa-ellipsis-v on card_icon"
                    id="toggleIcon"
                    role="presentation"
                />
                <div
                    className="card_title"
                >
                    Nairobi, Kenya
                     
                </div>
                <div
                    className="card_stipend"
                >
                    $
                    345
                     
                </div>
                <p
                    className="card_p"
                >
                    Daily Stipend
                </p>
            </div>
            <div
                className="card"
            >
                <i
                    className="fa fa-ellipsis-v on card_icon"
                    id="toggleIcon"
                    role="presentation"
                />
                <div
                    className="card_title"
                >
                    Kampala, Uganda
                     
                </div>
                <div
                    className="card_stipend"
                >
                    $
                    345
                     
                </div>
                <p
                    className="card_p"
                >
                    Daily Stipend
                </p>
            </div>
        </div>
    </div>
</div>,
]
`;
